crates/task-impls/src/helpers.rs

```diff
@@ -505,37 +507,39 @@ pub async fn validate_proposal_safety_and_liveness<TYPES: NodeType, V: Versions>
     // passes.
 
     // Liveness check.
-    let read_consensus = consensus.read().await;
-    let liveness_check = justify_qc.view_number() > read_consensus.locked_view();
-
-    // Safety check.
-    // Check if proposal extends from the locked leaf.
-    let outcome = read_consensus.visit_leaf_ancestors(
-        justify_qc.view_number(),
-        Terminator::Inclusive(read_consensus.locked_view()),
-        false,
-        |leaf, _, _| {
-            // if leaf view no == locked view no then we're done, report success by
-            // returning true
-            leaf.view_number() != read_consensus.locked_view()
-        },
-    );
-    let safety_check = outcome.is_ok();
-
-    ensure!(safety_check || liveness_check, {
-        if let Err(e) = outcome {
-            broadcast_event(
-                Event {
-                    view_number,
-                    event: EventType::Error { error: Arc::new(e) },
-                },
-                &event_sender,
-            )
-            .await;
-        }
+    {
```

Comment:
This bit me below, I just wrapped it so that way the consensus handles in this file go out of scope. It's on the roadmap to refactor a lot of this during the dependency refactor rollout.